src/main/java/org/commcare/cases/entity/AsyncNodeEntityFactory.java

```diff
@@ -78,7 +81,12 @@ protected void setEvaluationContextDefaultQuerySet(EvaluationContext ec,
      * Bulk loads search field cache from db.
      * Note that the cache is lazily built upon first case list search.
      */
-    private void primeCache() {
+    protected void primeCache() {
+        if (isCancelled) return;
+        if (progressListener != null) {
```

Comment:
Will it cause issues for this progress update to occur before all of the null checks below that dump out of the method and/or without a `finally {}` clause? It seems it it would keep the listener spinning forever.

Reply:
https://github.com/dimagi/commcare-core/pull/1454/commits/5b8b77bc057d7afae0e4ae15559daeb1b82898f6